### ISSUE TITLE
Problem with @Test(timeout=TIMEOUT)

### ISSUE THREAD

**Issue by cblanco** [tomakehurst/wiremock#492]
```
import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
import static com.github.tomakehurst.wiremock.client.WireMock.get;
import static com.github.tomakehurst.wiremock.client.WireMock.givenThat;
import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;

import com.github.tomakehurst.wiremock.junit.WireMockRule;

import org.junit.Rule;
import org.junit.Test;

public class DD {

    @Rule
    public WireMockRule wireMock = new WireMockRule(0);

    @Test(timeout=1000L)
    public void myT() {

        givenThat(get(urlEqualTo("/")).willReturn(aResponse().withBody("Hi!")));
        System.out.println("Hello");
    }
}
```

Using v1.57:
Test OK

Using v1.58, 2.1.11 or 2.1.12:

> org.apache.http.conn.HttpHostConnectException: Connect to localhost:8080 [localhost/127.0.0.1] failed: Connection refused
>   at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:151)
>   at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
>   at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:380)
>   at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
>   at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
>   at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
>   at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
>   at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:107)
>   at com.github.tomakehurst.wiremock.client.HttpAdminClient.safelyExecuteRequest(HttpAdminClient.java:266)
>   at com.github.tomakehurst.wiremock.client.HttpAdminClient.postJsonAssertOkAndReturnBody(HttpAdminClient.java:249)
>   at com.github.tomakehurst.wiremock.client.HttpAdminClient.addStubMapping(HttpAdminClient.java:115)
>   at com.github.tomakehurst.wiremock.client.WireMock.register(WireMock.java:213)
>   at com.github.tomakehurst.wiremock.client.WireMock.register(WireMock.java:209)
>   at com.github.tomakehurst.wiremock.client.WireMock.givenThat(WireMock.java:80)
>   at DD.myT(DD.java:20)
>   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>   at java.lang.reflect.Method.invoke(Method.java:498)
>   at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
>   at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
>   at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
>   at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
>   at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
>   at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
>   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
>   at java.lang.Thread.run(Thread.java:745)
> Caused by: java.net.ConnectException: Connection refused
>   at java.net.PlainSocketImpl.socketConnect(Native Method)
>   at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
>   at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
>   at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
>   at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
>   at java.net.Socket.connect(Socket.java:589)
>   at org.apache.http.conn.socket.PlainConnectionSocketFactory.connectSocket(PlainConnectionSocketFactory.java:74)
>   at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:134)
>   ... 26 more

The problem does not occur by calling givenThat using wireMock.givenThat(...)

**Comment by tomakehurst** [comment]
This is a tricky one. The problem is that JUnit puts the test into a different thread than the rule, and since a ThreadLocal is used to bind the static DSL to the rule, it can't find it. Not sure there's a workaround for this, other than using the instance method as you've suggested.

**Comment by tomakehurst** [comment]
Fixed in 6045de537c7dbc76b23e6a93829222a30735a11d
